mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go

```diff
@@ -60,13 +59,8 @@ func TestAccConfigRSAlertConfiguration_EmptyMetricThresholdConfig(t *testing.T)
 		Steps: []resource.TestStep{
 			{
 				Config: testAccMongoDBAtlasAlertConfigurationConfigEmptyMetricThresholdConfig(orgID, projectName, true),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckMongoDBAtlasAlertConfigurationExists(resourceName, alert),
-					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttr(resourceName, "notification.#", "1"),
-					resource.TestCheckResourceAttrSet(resourceName, "threshold_config.#"),
-				),
-				ExpectNonEmptyPlan: true,
+				// metric threshold is not created so state is inconsistent with plan
+				ExpectError: regexp.MustCompile("Error: Provider produced inconsistent result after apply"),
```

Comment:
Are we planning to handle this error? Or are we waiting for a response from Hashicorp on this? 
Migration tests are basically to ensure backward compatibility so this would be a breaking change.

Reply:
This acceptance test defines an an empty block `metric_threshold_config {}` which does not have an initial validation as as all attributes are optional. 
- In 1.11.0 the apply runs successfully, but in the terraform state `metric_threshold_config` is an empty array. That is why when running terraform plan afterwards you receive a non empty plan as terraform mentions that this block needs to be created.
- With the new framework, after running terraform apply the output state is the same, with `metric_threshold_config` as an empty array. However the terraform apply responds with an error because the framework now validates that the state is inconsistent with the defined configuration.

I am not that concerned about this case as it is essentially an incorrectly defined configuration that users would fix when using 1.11.0